packages/shared/src/common/retry.ts

```diff
@@ -3,17 +3,18 @@ import { MetriportError } from "../error/metriport-error";
 import { errorToString } from "../error/shared";
 import { sleep } from "./sleep";
 
-export const defaultOptions: Required<ExecuteWithRetriesOptions> = {
+export const defaultOptions: Required<ExecuteWithRetriesOptions<unknown>> = {
   initialDelay: 10,
   maxDelay: Infinity,
   backoffMultiplier: 2,
   maxAttempts: 10,
   shouldRetry: () => true,
+  shouldRetryResult: () => false,
```

Comment:
this needs to default to false otherwise it will infinite loop and retry forever if no function is passed to it.